cmd/root.go

```diff
@@ -147,13 +150,22 @@ var rootCmd = &cobra.Command{
 			deprecatedVersionList = defaultVersions
 		}
 
-		// From this compiled list of deprecations, build a component list
+		// From the compiled list of deprecations and the components flag, build a component list
 		var componentList []string
 		for _, v := range deprecatedVersionList {
-			if !stringInSlice(v.Component, componentList) {
-				componentList = append(componentList, v.Component)
+			if !utils.StringInSlice(v.Component, componentList) {
+				if components != nil {
```

Comment:
I think we could get rid of this if statement if, on line 67, we set the default to an empty list instead of nil. Unsure if that works the way I think in practice but worth trying.

Reply:
The preview shows two if statements, but I'm talking about the one on line 157.

Reply:
I think that is possible. I'll try it out.

Reply:
Okay, so looking closer, it doesn't really work like that.

We have to check if the flag is nil or zero-length so that we can decide whether to use it or the generated list. 

The logic there is a little confusing at first, and re-visiting took me way too long to remember why I wrote it like that. I'll see if there's a way to make it less confusing or at least add comments

Reply:
No worries, I get what it's doing I was just hoping we could get rid of one layer of if statements. Actually wonder now if it would work if the default sent via cobra on line 67 is simply `""` since cobra is the mechanism that turns it into a slice.

Reply:
For posterity, you can't set the default to `""` you have to set it to `[]string{}`.